macOS failed due to: 

```
fatal: unable to parse config file '/Users/buildnode/jenkins/workspace/swift-PR-osx/branch-master/swift/test/lit.cfg', traceback: Traceback (most recent call last):
  File "/Users/buildnode/jenkins/workspace/swift-PR-osx/branch-master/llvm/utils/lit/lit/TestingConfig.py", line 88, in load_from_path
    exec(compile(data, path, 'exec'), cfg_globals, None)
  File "/Users/buildnode/jenkins/workspace/swift-PR-osx/branch-master/swift/test/lit.cfg", line 927, in <module>
    config.available_features.remove('tsan_runtime')
KeyError: 'tsan_runtime'

*** Failed while running tests for swift (check-swift-validation-iphonesimulator-i386)
```

--- a/test/lit.cfg
+++ b/test/lit.cfg
@@ -908,17 +908,24 @@ def source_compiler_rt_libs(path):
 source_compiler_rt_libs(os.path.join(test_resource_dir, 'clang', 'lib',
                         platform.system().lower()))
 
-def check_runtime_feature(name):
+def check_runtime_libs(features_to_check):
     for lib in config.compiler_rt_libs:
-        if lib.startswith('libclang_rt.' + name + '_'):
-            config.available_features.add(name + '_runtime')
-            return
+        for (libname, feature) in features_to_check:
+            if lib.startswith("libclang_rt." + libname + "_"):
+                config.available_features.add(feature)
 
-check_runtime_feature('profile')
-check_runtime_feature('asan')
-check_runtime_feature('ubsan')
-check_runtime_feature('tsan')
-check_runtime_feature('safestack')
+runtime_libs = [
+    ('profile', 'profile_runtime'),
+    ('asan', 'asan_runtime'),
+    ('ubsan', 'ubsan_runtime'),
+    ('tsan', 'tsan_runtime'),
+    ('safestack', 'safestack_runtime')
+]
+
+if run_ptrsize != "32":
+    runtime_libs.append(('tsan', 'tsan_runtime'))
+
+check_runtime_libs(libs_to_check)
 
 if not getattr(config, 'target_run_simple_swift', None):
     config.target_run_simple_swift = (

--- a/test/lit.cfg
+++ b/test/lit.cfg
@@ -925,7 +925,7 @@ runtime_libs = [
 if run_ptrsize != "32":
     runtime_libs.append(('tsan', 'tsan_runtime'))
 
-check_runtime_libs(libs_to_check)
+check_runtime_libs(runtime_libs)
 
 if not getattr(config, 'target_run_simple_swift', None):
     config.target_run_simple_swift = (

